services/discovery/server_test.go

```diff
@@ -0,0 +1,73 @@
+package discovery_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	commonpb "go.viam.com/api/common/v1"
+	pb "go.viam.com/api/service/discovery/v1"
+	"go.viam.com/test"
+	"go.viam.com/utils/protoutils"
+
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/services/discovery"
+	"go.viam.com/rdk/testutils"
+	"go.viam.com/rdk/testutils/inject"
+)
+
+var errDoFailed = errors.New("do failed")
+
+func newServer() (pb.DiscoveryServiceServer, *inject.DiscoveryService, *inject.DiscoveryService, error) {
+	injectDiscovery1 := &inject.DiscoveryService{}
+	injectDiscovery2 := &inject.DiscoveryService{}
+	resourceMap := map[resource.Name]resource.Resource{
+		discovery.Named(testDiscoveryName): injectDiscovery,
+		discovery.Named(failDiscoveryName): injectDiscovery2,
+	}
+	injectSvc, err := resource.NewAPIResourceCollection(discovery.API, resourceMap)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	return discovery.NewRPCServiceServer(injectSvc).(pb.DiscoveryServiceServer), injectDiscovery, injectDiscovery2, nil
+}
+
+func TestDiscoveryDo(t *testing.T) {
```

Comment:
since this is literally copy/paste from the generic service I should probably remove this test, but I'm also fine with leaving it in case we are worried this will somehow break